pinot-segment-local/src/main/java/org/apache/pinot/segment/local/recordtransformer/ExpressionTransformer.java

```diff
@@ -40,12 +41,18 @@
  */
 public class ExpressionTransformer implements RecordTransformer {
 
-  private final LinkedHashMap<String, FunctionEvaluator> _expressionEvaluators = new LinkedHashMap<>();
+  @VisibleForTesting
+  final LinkedHashMap<String, FunctionEvaluator> _expressionEvaluators = new LinkedHashMap<>();
 
   public ExpressionTransformer(TableConfig tableConfig, Schema schema) {
     Map<String, FunctionEvaluator> expressionEvaluators = new HashMap<>();
     if (tableConfig.getIngestionConfig() != null && tableConfig.getIngestionConfig().getTransformConfigs() != null) {
       for (TransformConfig transformConfig : tableConfig.getIngestionConfig().getTransformConfigs()) {
+        String columnName = transformConfig.getColumnName();
+        if (expressionEvaluators.containsKey(columnName)) {
+          throw new RuntimeException("Cannot set more than one ingestion transform function on a column (column name: "
+              + "'" + columnName + "').");
+        }
```

Comment:
This part can be merged into the put to avoid one extra lookup
```suggestion
        FunctionEvaluator previous = expressionEvaluators.put(transformConfig.getColumnName(),
              FunctionEvaluatorFactory.getExpressionEvaluator(transformConfig.getTransformFunction()));
        Preconditions.checkState(previous == null, "Cannot set more than one ingestion transform function on column: %s", transformConfig.getColumnName());
```

Reply:
Fixed.